js/setpassword.js

```diff
@@ -0,0 +1,69 @@
+SetPassword = {
```

Comment:
this would define a symbol `SetPassword` on the global namespace, it's worse than before. check with the debugger and see that `window.SetPassword` is now defined.

To avoid this, see a pattern like https://github.com/owncloud/core/blob/master/apps/files/js/filelist.js#L36. You need to wrap this in an anonymous function so it is scoped only locally, like here https://github.com/owncloud/core/blob/master/apps/files/js/filelist.js#L11.

Or alternatively just write `OCA.UserManagement.SetPassword = {` on this line... make sure the namespace `OCA.UserManagement` exists before this runs.

Reply:
Updated the change. Tried to use anonymous function to wrap `SetPassword`. And then tried to set `OCA.UserManagement`. Let me know the change looks ok.